src/components/KFileUpload/KFileUpload.vue

```diff
@@ -0,0 +1,344 @@
+<template>
+  <div
+    class="k-file-upload w-100"
+  >
+    <KLabel
+      v-if="label && appearance === 'file'"
+      :for="customInputId"
+      v-bind="labelAttributes"
+      data-testid="k-file-upload-label"
+    >
+      {{ label }}
+    </KLabel>
+
+    <KInput
+      :id="customInputId"
```

Comment:
question: why do we need to bind an `id` attribute to the input? This causes issues when the id changes if not provided in consuming apps (e.g. snapshot testing).

Reply:
This is so we can use `for` attribute in the label which we definitely should do

Reply:
Yes, I meant to delete this comment. I don't like the generated ids because they cause unpredictable changes in the component if a consuming app does snapshot testing, but we're already doing this elsewhere